### PR TITLE
feat(transaction-summary): Add analytics to track biases sufficiency assessment

### LINKED ISSUE #47138: Biases Sufficiency Assessment
The goal of this ticket is to instrument the sentry application in a way that gives us a good signal of whether our sampling biases are doing a good job.

Basically, whenever we have a hole in the data (not enough samples), we want to fire an event.

A good place to track this would be on the Transaction Summary page.

Image

Here are a few things to keep in mind:
- we need a baseline so fire an event every time the Overview tab of the transaction summary is loaded
  - fire this only if the query can be fulfilled with metrics (the API response that powers the chart on top has an isMetrics boolean flag on it)
  - initially implement just for the Duration Breakdown visualization but in case there the page triggers another request that can be used, use that instead.
- fire a different event when the events table is empty
  - again, do this only if the query can be fulfilled with metrics
- we probably want to do all of this only for AM2 orgs with a plan above 5M transactions; otherwise, we would be heavily skewing numbers in our favor

### PR DESCRIPTION
**Pull request by priscilawebdev**
closes https://github.com/getsentry/sentry/issues/47138

### CODE REVIEW

File: static/app/types/organization.tsx [hunk 1]
```diff
@@ -41,7 +41,8 @@ export interface Organization extends OrganizationSummary {
   allowJoinRequests: boolean;
   allowSharedIssues: boolean;
   attachmentsRole: string;
-  availableRoles: {id: string; name: string}[]; // Deprecated, use orgRoleList
+  availableRoles: {id: string; name: string}[];
+  // Deprecated, use orgRoleList
```
Comment (priscilawebdev) [inline static/app/types/organization.tsx]: hm this was done by the eslint

File: static/app/views/performance/transactionSummary/transactionOverview/index.tsx [hunk 2]
```diff
@@ -149,6 +149,22 @@ function OverviewContentWrapper(props: ChildProps) {
     error: totalCountQueryError,
   } = totalCountQueryData;
 
+  const isTableEmpty = !isLoading && !error && tableData.data.length === 0;
+
+  useEffect(() => {
+    if (mepContext.isMetricsData && organization.isDynamicallySampled) {
+      trackAnalytics('dynamic_sampling_transaction_summary.baseline', {
+        organization,
```
Comment (priscilawebdev) [inline static/app/views/performance/transactionSummary/transactionOverview/index.tsx:157]: @matejminar is `organization` enough or do we want to gather other info?

File: static/app/utils/analytics/dynamicSamplingAnalyticsEvents.tsx [hunk 3]
```diff
@@ -9,11 +9,17 @@ export type DynamicSamplingEventParameters = {
     id: DynamicSamplingBiasType;
     project_id: string;
   };
+  'dynamic_sampling_transaction_summary.baseline': {};
+  'dynamic_sampling_transaction_summary.no_events': {};
```
Comment (matejminar) [inline static/app/utils/analytics/dynamicSamplingAnalyticsEvents.tsx:13]:
```suggestion
  'dynamic_sampling_transaction_summary.no_samples': {};
```

File: static/app/views/performance/transactionSummary/transactionOverview/index.tsx (hunk 2 above)
Comment (matejminar) [reply static/app/views/performance/transactionSummary/transactionOverview/index.tsx:157]: We get the URL for free, which should be enough (although query params might be missing to precisely locate the gap). Let's merge and see.

**Review by matejminar — COMMENTED** [review]
image
We should fire `no_samples` event on top of `baseline` event in this scenario.


image
We should not fire `baseline` if chart is empty.

**Comment by priscilawebdev** [comment]
checking tests...

**Comment by priscilawebdev** [comment]
checking issue with the acceptance tests...

**Comment by matejminar** [comment]
We should not fire `no_samples` if there are no metrics results :(
image

**Review by matejminar — APPROVED** [review]

**Pull request merged**